store/postgres/src/relational/prune.rs

```diff
@@ -311,20 +352,17 @@ impl Layout {
         // Copy nonfinal entities, and replace the original `src` table with
         // the smaller `dst` table
         reporter.start_switch();
-        conn.transaction(|| -> Result<(), CancelableError<StoreError>> {
-            //  see also: deployment-lock-for-update
-            deployment::lock(conn, &self.site)?;
-
+        //  see also: deployment-lock-for-update
+        deployment::with_lock(conn, &self.site, || -> Result<_, StoreError> {
             for table in &prunable_tables {
                 reporter.copy_nonfinal_start(table.src.name.as_str());
-                let rows = table.copy_nonfinal_entities(conn, final_block)?;
-                reporter.copy_nonfinal_finish(table.src.name.as_str(), rows);
-                cancel.check_cancel()?;
+                table.copy_nonfinal_entities(conn, reporter, final_block)?;
+                cancel.check_cancel().map_err(CancelableError::from)?;
             }
 
             for table in prunable_tables {
-                table.switch(conn, self)?;
-                cancel.check_cancel()?;
+                conn.transaction(|| table.switch(conn, self))?;
+                cancel.check_cancel().map_err(CancelableError::from)?;
```

Comment:
I _think_ it was because before the closure was declared as returning an error `CancelableError<StoreError>` of and after the change the typechecker needed this hint to find the conversions from `Canceled -> CancelableError -> StoreError`